Add date range download dialog

lcsfelix opened the pull request.

lcsfelix committed:
feat: add date range download dialog

RafaelTaranto commented on new-lamassu-admin/package.json:
```diff
@@ -15,6 +15,7 @@
     "jss-plugin-extend": "^10.0.0",
     "lodash": "4.17.15",
     "moment": "2.24.0",
+    "moment-range": "^4.0.2",
```
I'm not seeing this being used anywhere

RafaelTaranto commented on new-lamassu-admin/src/components/date-range-picker/Calendar.js:
```diff
@@ -0,0 +1,144 @@
+import { ReactComponent as Arrow } from '../../styling/icons/arrow/month_change.svg'
+import { primaryColor, zircon, fontSecondary } from '../../styling/variables'
+import typographyStyles from '../typography/styles'
+const { label2 } = typographyStyles
```
We should stablish a styleguide for imports organization. I like to do it this way, separating types with a whitespace.

```js
# imports from node-modules
import React, { useState } from 'react'
import moment from 'moment'
import { toInteger } from 'lodash/fp'
import { makeStyles } from '@material-ui/core/styles'

# imports from components outside of ./
import { ReactComponent as Arrow } from '../../styling/icons/arrow/month_change.svg'
import { primaryColor, zircon, fontSecondary } from '../../styling/variables'
import typographyStyles from '../typography/styles'

# imports from this components folder
import Tile from './Tile'

# not imports
const { label2 } = typographyStyles
```

RafaelTaranto commented on new-lamassu-admin/src/components/date-range-picker/Calendar.js:
```diff
@@ -0,0 +1,144 @@
+    display: 'flex',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    padding: [[15, 15]],
```
pretty cool that this works, I have to adapt my styles and stop using string interpolation for this

RafaelTaranto commented on new-lamassu-admin/src/components/date-range-picker/Calendar.js:
```diff
@@ -0,0 +1,144 @@
+    },
+    fontFamily: fontSecondary,
+    fontSize: 16,
+    fontWeight: 500
```
this should be extended from some predefined typography

lcsfelix replied:
The reasoning behind this and https://github.com/lamassu/lamassu-server/pull/328#discussion_r345377519 was that they are both specific things inside this container that shouldn't really be changed globally (more specifically the font size).

RafaelTaranto commented on new-lamassu-admin/src/components/date-range-picker/DateRangePicker.js:
```diff
@@ -0,0 +1,131 @@
+import { primaryColor, offColor, zircon } from '../../styling/variables'
+
+import typographyStyles from '../typography/styles'
+const { info1, label, label3 } = typographyStyles
```
same as the other comment

RafaelTaranto commented on new-lamassu-admin/src/components/date-range-picker/Tile.js:
```diff
@@ -0,0 +1,94 @@
+    zIndex: 2,
+    fontFamily: fontSecondary,
+    fontSize: 14,
+    fontWeight: 500
```
Same as the other one, should be extended from some typography style.

RafaelTaranto commented on new-lamassu-admin/src/components/date-range-picker/Tile.js:
```diff
@@ -0,0 +1,94 @@
+    fontWeight: 500
+  },
+  lowerBound: {
+    width: [['50%', '!important']],
```
using !important should be prevented, I'll take a look at why you're needing to do this and detail this comment further.

lcsfelix replied:
It's so it doesn't overflow when the bounds are on Saturdays/Sundays.
![image](https://user-images.githubusercontent.com/5789328/68760370-973c3880-0609-11ea-9472-527029356af7.png)
I can replace it with an `overflow: hidden` on the wrapper, if that's preferred.

lcsfelix replied:
I also noticed a typo, should be `[['50%'], '!important']`.

RafaelTaranto replied:
Yeah, if overflow hidden on the wrapper works that's preferred.

RafaelTaranto commented on new-lamassu-admin/src/components/typography/styles.js:
```diff
@@ -119,6 +119,11 @@ export default {
+    fontSize: fontSize5,
+    fontFamily: fontSecondary,
+    fontWeight: 700
+  },
```
This was already on the codebase, not sure why it's showing up here again

RafaelTaranto commented on new-lamassu-admin/src/pages/ServerLogs.js:
```diff
@@ -55,6 +57,40 @@ styles.uptimeContainer = {
+styles.download = {
+  padding: [[30, 15, 30, 15]]
+}
+
```
Why aren't these on the style files?

lcsfelix replied:
Resolved in https://github.com/lamassu/lamassu-server/pull/328/commits/1dbb47aa94c828e4033ab65b63bb6fa3f4a970fc.

lcsfelix committed:
fix: reorganize imports

lcsfelix committed:
refactor: move download popover to own component

lcsfelix committed:
fix: apply styles

lcsfelix committed:
refactor: move selected date container up

Allows reuse of the date range picker by making the display of the
selected range independent from the picker.

lcsfelix committed:
fix: make Calendar method more fp like

joshmh commented on new-lamassu-admin/src/components/date-range-picker/DateRangePicker.js:
```diff
@@ -0,0 +1,50 @@
+      if (day.isBefore(from, 'day')) {
+        setTo(from)
+        setFrom(day)
+      } else {
```
Use ``return``s instead of ``else``s.

joshmh commented on new-lamassu-admin/src/components/date-range-picker/Tile.js:
```diff
@@ -0,0 +1,92 @@
+
+const styles = {
+  wrapper: {
+
```
Remove blank line.

lcsfelix committed:
fix: use returns instead of elses

lcsfelix commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Create maintenance logs download date range picker](https://trello.com/c/Hz9fj7UB/114-create-maintenance-logs-download-date-range-picker)
